flink-runtime/src/main/java/org/apache/flink/runtime/rpc/taskexecutor/TaskExecutor.java

```diff
@@ -94,12 +729,11 @@ public void notifyOfNewResourceManagerLeader(String newLeaderAddress, UUID newLe
 			if (newLeaderAddress != null) {
 				// the resource manager switched to a new leader
 				log.info("ResourceManager leader changed from {} to {}. Registering at new leader.",
-						resourceManagerConnection.getResourceManagerAddress(), newLeaderAddress);
```

Comment:
Why are these essentials RPC methods at the bottom. They should be moved to the top of the class.